javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -195,6 +195,8 @@ final class Matcher private[regex] (
 
   def group(group: Int): String = ensureLastMatch(group).orNull
 
+  def group(name: String): String = throw new NoSuchElementException
```

Comment:
Er ... It's supposed to return an `IllegalArgumentException`: https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#group%28java.lang.String%29

Also ideally it should throw an `IllegalStateException` instead if no match was attempted so far (like calling `ensureLastMatch`).